OpenRA.Test/OpenRA.Game/CoordinateTest.cs

```diff
@@ -10,32 +10,82 @@
 #endregion
 
 using System;
+using System.Linq;
 using NUnit.Framework;
 
 namespace OpenRA.Test
 {
 	[TestFixture]
 	public class CoordinateTest
 	{
-		[TestCase(TestName = "Test CPos and MPos conversion and back again.")]
-		public void CoarseToMapProjection()
+		[TestCase(TestName = "Test CPos to MPos conversion and back again.")]
+		public void CoarseToMapProjection1()
 		{
 			foreach (var gridType in Enum.GetValues<MapGridType>())
 			{
-				for (var x = 0; x < 12; x++)
-				{
-					for (var y = 0; y < 12; y++)
-					{
-						var cell = new CPos(x, y);
-
-						// Known problem on isometric mods that shouldn't be visible to players as these are outside the map.
-						if (gridType == MapGridType.RectangularIsometric && y > x)
-							continue;
-
-						Assert.That(cell, Is.EqualTo(cell.ToMPos(gridType).ToCPos(gridType)));
-					}
-				}
+				var original = new CellCoordsRegion(new CPos(-12, -12), new CPos(12, 12));
+				var expected = original.Select(pos => pos.ToMPos(gridType).ToCPos(gridType)).ToArray();
+
+				Assert.That(original, Is.EqualTo(expected));
 			}
 		}
+
+		[TestCase(TestName = "Test MPos to CPos conversion and back again.")]
+		public void CoarseToMapProjection2()
```

Comment:
These test names should probably also be updated? I'd suggest `CPosConversionRoundtrip` and `MPosConversionRoundtrip`.